js/web/components/Subscribe.js

```diff
@@ -0,0 +1,134 @@
+  const {accountAddress, hubHandle, inFeed} = props;
+  const wallet = useWallet()
+  const router = useRouter()
+  const { subscriptionSubscribe, subscriptionUnsubscribe, userSubscriptions, getSubscriptionsForUser, subscriptionState } = useContext(Nina.Context)
+  const [isFollowing, setIsFollowing] = useState(false)
+  const [pending, setPending] = useState(false)
+  const [targetSubscriptions, setTargetSubscriptions] = useState()
+  const [followsYou, setFollowsYou] = useState(false)
+  const [isUser, setIsUser] = useState(false)
+  const [inDashboard, setInDashboard] = useState(router.pathname.includes('/dashboard'))
+  const { enqueueSnackbar } = useSnackbar()
+
+
+  useEffect(() => {
+    const handleSubscriptions = async (accountAddress) => {
+      const subscriptions = await getSubscriptionsForUser(accountAddress)
+      setTargetSubscriptions(subscriptions)
+    }
+    handleSubscriptions(accountAddress)
+  }, [])
+
+  useEffect(() => {
+    if (userSubscriptions) {
+      const checkIfFollowing = userSubscriptions.some(subscription => subscription.to === accountAddress)
+      setIsFollowing(checkIfFollowing)
+    }
+  }, [userSubscriptions])
+
+  useEffect(() => {
+    if (wallet.publicKey) {
+      setIsUser(wallet.publicKey.toBase58() === accountAddress)
+    }
+    if (wallet.connected && targetSubscriptions) {
+      const checkIfFollowsYou = targetSubscriptions?.some(subscription => subscription.to === wallet.publicKey.toBase58())
+      setFollowsYou(checkIfFollowsYou)
+    }
+  }, [targetSubscriptions, wallet.connected])
+
+  const handleSubscribe = async (accountAddress, hubHandle) => {
+    setPending(true)
+    const result = await subscriptionSubscribe(accountAddress, hubHandle)
+    console.log('result', result)
+    if (result.success) {
+      enqueueSnackbar(result.msg, {
+        variant: 'success',
+      })
+    } else {
+      enqueueSnackbar('Error Following Account.', {
+        variant: 'error',
+      })
+    }
+    setPending(false)
+  }
+
+  const handleUnsubscribe = async (accountAddress) => {
+    setPending(true)
+    const result = await subscriptionUnsubscribe(accountAddress)
+    if (result.success) {
+      enqueueSnackbar(result.msg, {
+        variant: 'success',
+      })
+    } else {
+      enqueueSnackbar('Error Unfollowing Account.', {
+        variant: 'error',
+      })
+    }
+    setPending(false)
+  }
+
+  return (
+    <Box display='flex' alignItems={'center'}>
+      {pending && (
+        <Box sx={{ padding: '0 15px'}}>
+          <Dots />
+        </Box>
+      )}
+
+      {(false) ? (
+        <Box sx={{ padding: '0 15px'}}>
+          {!inDashboard && (
+            <Typography>This is how your profile appears to to other users. Click <Link href="/dashboard"><a style={{textDecoration: 'underline'}}>here</a></Link> to see your dashboard</Typography>
+          )}
+          {inDashboard && (
+            <Typography>Welcome to your dashboard</Typography>
+          )}
```

Comment:
is this necessary? im not sure it is, but if it is it prob doesn't belong in subscribe component